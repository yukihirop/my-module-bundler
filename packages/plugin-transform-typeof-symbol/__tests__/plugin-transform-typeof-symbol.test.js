'use strict';

const pluginTransformTypeofSymbol = require('..');

describe('plugin-transform-typeof-symbol', () => {
    it('needs tests');
});
