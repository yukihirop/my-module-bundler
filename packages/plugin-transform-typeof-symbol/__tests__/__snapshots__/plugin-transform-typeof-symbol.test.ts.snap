// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typeof-symbol basic 1`] = `
"(function(modules) {
  function require(id) {
    const [fn, mapping] = modules[id];

    function localRequire(name) {
      return require(mapping[name]);
    }

    const module = {
      exports: {}
    };

    fn(localRequire, module, module.exports);

    return module.exports;
  }

  require(0)
})({
  0: [
    function(require, module, exports) {
      function _typeof(obj) {
        if (typeof Symbol === \\"function\\" && typeof Symbol.iterator === \\"symbol\\") {
          _typeof = function _typeof(obj) {
            return typeof obj;
          };
        } else {
          _typeof = function _typeof(obj) {
            return obj && typeof Symbol === \\"function\\" && obj.constructor === Symbol && obj !== Symbol.prototype ? \\"symbol\\" : typeof obj;
          };
        }

        return _typeof(obj);
      }

      _typeof({}) === \\"object\\";
    },
    {},
  ]
})"
`;
